test/e2e/azure_test.go

```diff
@@ -350,4 +351,60 @@ var _ = Describe("Workload cluster creation", func() {
 			})
 		})
 	})
+
+	Context("Creating a cluster using a different SP identity", func() {
+		BeforeEach(func() {
```

Comment:
Same comment I wrote in the Windows PR - we need to audit all the clusters that get created in pr e2e tests and see if any of them can be safely combined to avoid growing the cost and duration of each PR run too much

Reply:
I think it makes sense for this test to have its own cluster since it using a different identity which we would want to test separately, wdyt?